aiosmtpd/smtp.py

```diff
@@ -535,14 +543,9 @@ def smtp_DATA(self, arg):
         received_data = EMPTYBYTES.join(data)
         if self._decode_data:
             received_data = received_data.decode('utf-8')
-        args = (self.peer, self.mailfrom, self.rcpttos, received_data)
-        kwargs = {}
-        if not self._decode_data:
-            kwargs = {
-                'mail_options': self.mail_options,
-                'rcpt_options': self.rcpt_options,
-                }
-        kwargs.update({'loop': self.loop})
+        self.message.received_data = received_data
+        args = (self.session, self.message)
+        kwargs = {'loop': self.loop}
```

Comment:
Why? It looks like session related information. I kept it alone because I put it under PR #41 scope - there it is a part of handler object. But I just remember that there can be strange cases when handler and protocol works in different loops, so may be pu it on session is not bad idea.